build.ps1

```diff
@@ -245,14 +246,20 @@ try {
         Write-Host -ForegroundColor Cyan "Unit Test execution"
         Write-Host
 
-        RunTests iothub\device\tests "IoT Hub DeviceClient Tests"
-        RunTests iothub\service\tests "IoT Hub ServiceClient Tests"
-        RunTests provisioning\device\tests "Provisioning Device Client Tests"
-        RunTests provisioning\transport\amqp\tests "Provisioning Transport for AMQP Tests"
-        RunTests provisioning\transport\http\tests "Provisioning Transport for HTTP Tests"
-        RunTests provisioning\transport\mqtt\tests "Provisioning Transport for MQTT Tests"
-        RunTests security\tpm\tests "SecurityProvider for TPM Tests"
-        RunTests provisioning\service\tests "Provisioning Service Client Tests"
+        $framework="*"
+        if (-not (IsWindowsDesktopDevelopmentBox))
+        {
+            $framework="netcoreapp2.1"
+        }
+
+        RunTests iothub\device\tests "IoT Hub DeviceClient Tests" $framework
```

Comment:
This is not necessary; unit tests are set up to always run against netcoreapp2.1 TFM:
https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/tests/Microsoft.Azure.Devices.Client.Test.csproj#L4
https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/service/tests/Microsoft.Azure.Devices.Test.csproj#L4
https://github.com/Azure/azure-iot-sdk-csharp/blob/master/provisioning/device/tests/Microsoft.Azure.Devices.Provisioning.Client.UnitTests.csproj#L4
https://github.com/Azure/azure-iot-sdk-csharp/blob/master/provisioning/service/tests/Microsoft.Azure.Devices.Provisioning.Service.Test.csproj#L4
https://github.com/Azure/azure-iot-sdk-csharp/blob/master/provisioning/transport/amqp/tests/Microsoft.Azure.Devices.Provisioning.Transport.Amqp.UnitTests.csproj#L4
https://github.com/Azure/azure-iot-sdk-csharp/blob/master/provisioning/transport/mqtt/tests/Microsoft.Azure.Devices.Provisioning.Transport.Mqtt.UnitTests.csproj#L4
https://github.com/Azure/azure-iot-sdk-csharp/blob/master/provisioning/transport/http/tests/Microsoft.Azure.Devices.Provisioning.Transport.Http.UnitTests.csproj#L4
https://github.com/Azure/azure-iot-sdk-csharp/blob/master/security/tpm/tests/Microsoft.Azure.Devices.Provisioning.Security.Tpm.UnitTests.csproj#L4

Reply:
I see a different behavior. If the _framework_ parameter is not explicitly specified (by default it is _"*"_), I see failures where _RunTests_ (more specifically the _"dontnet test"_ command) is trying to build for the full .Net Framework.

Reply:
I would imagine the behavior to be the same as what we expect for Linux systems (Linux systems cannot run apps targeting .NET framework either). Can you send me a snapshot of the error logs?

Reply:
I think you are right here, explicitly specifying the framework does not seem to be needed.
In the log file (https://azure-iot-sdks.visualstudio.com/azure-iot-sdks/_build/results?buildId=42255) I do not see issues due to .Net Framework. I may have hit some other issue when running on a local device.